packages/react-server/src/ReactFizzServer.js

```diff
@@ -2139,8 +2159,7 @@ function flushCompletedQueues(
     }
     largeBoundaries.splice(0, i);
   } finally {
-    completeWriting(destination);
-    flushBuffered(destination);
+    let allComplete = false;
```

Comment:
You can avoid this flag and just use control flow here instead. Just duplicate the completeWriting/flushBuffered in the else branch.

Reply:
What's the tradeoff here? Less branching but trivially more code?